src/lib/SequentialBucket.ts

```diff
@@ -0,0 +1,52 @@
+type RatelimitedFunction = () => Promise<void>;
+
+const wait = (ms: number) => new Promise((rs) => setTimeout(rs, ms));
+
+// TODO(TTtie): test this in the real world
+class SequentialBucket {
```

Comment:
What purpose does this serve that the normal bucket does not?

Reply:
this bucket calls async functions async-ly and doesn't statically decrease the remaining bucket tokens unlike the regular one does